test/api/route/messages_test.dart

```diff
@@ -608,6 +608,79 @@ void main() {
     });
   });
 
+  group('getFileTemporaryUrl', () {
+    test('constructs URL correctly from response', () {
+      return FakeApiConnection.with_((connection) async {
+        connection.prepare(json: {
+          'url': '/user_uploads/temporary/abc123',
+          'result': 'success',
+          'msg': '',
+        });
+
+        final result = await getFileTemporaryUrl(connection,
+          filePath: '/user_uploads/1/2/testfile.jpg');
+
+        check(result.toString()).equals('${connection.realmUrl}/user_uploads/temporary/abc123');
+        check(connection.lastRequest).isA<http.Request>()
+          ..method.equals('GET')
+          ..url.path.equals('/api/v1/user_uploads/1/2/testfile.jpg');
+      });
+    });
+
+  test('returns temporary URL for valid realm file', () {
+    return FakeApiConnection.with_((connection) async {
+      connection.prepare(json: {
+        'url': '/user_uploads/temporary/abc123',
+        'result': 'success',
+        'msg': '',
+      });
+
+      final result = await tryGetFileTemporaryUrl(connection,
+        url: Uri.parse('${connection.realmUrl}user_uploads/123/testfile.jpg'),
+        realmUrl: connection.realmUrl);
+
+      check(result).isNotNull();
+      check(result.toString()).equals('${connection.realmUrl}/user_uploads/temporary/abc123');
+    });
+  });
+
+  test('returns null for non-realm URL', () {
+    return FakeApiConnection.with_((connection) async {
+      final result = await tryGetFileTemporaryUrl(connection,
+        url: Uri.parse('https://example.com/image.jpg'),
```

Comment:
```suggestion
        url: Uri.parse('https://example.com/user_uploads/123/testfile.jpg'),
```
The URL provided is both non-realm URL and a non-matching URL, but the test case is only about non-realm URL, so it's good to adjust it to only what the test case is about. That way, it's easy to validate it against what the test case claims.